addons/sale_loyalty/views/sale_order_views.xml

```diff
@@ -22,12 +22,27 @@
                 <button name="%(sale_loyalty.sale_loyalty_coupon_wizard_action)d"
                         type="action"
                         string="Coupon Code"
-                        class="btn btn-secondary"/>
+                        class="btn btn-secondary"
+                        invisible="locked or state == 'cancel'"/>
+                <button name="%(sale_loyalty.sale_loyalty_coupon_wizard_action)d"
+                        type="action"
+                        string="Coupon Code"
+                        class="btn btn-secondary"
+                        disabled="1"
+                        invisible="not (locked or state == 'cancel')"/>
```

Comment:
```suggestion
                <button
                    string="Coupon Code"
                    name="%(sale_loyalty.sale_loyalty_coupon_wizard_action)d"
                    type="action"
                    class="btn btn-secondary"
                    disabled="1"
                    invisible="not (locked or state == 'cancel')"
                />
```

Suggestions:
- Attribute order
- little better indent

Appy same in other buttons also